app/dv/src/main/java/io/syndesis/dv/lsp/parser/statement/FromClause.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.dv.lsp.parser.statement;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.lsp4j.Position;
+import org.teiid.query.parser.Token;
+
+import io.syndesis.dv.lsp.parser.DdlAnalyzerException;
+import io.syndesis.dv.lsp.parser.DdlTokenAnalyzer;
+
+public class FromClause extends AbstractStatementObject {
+
+    public FromClause(DdlTokenAnalyzer analyzer) {
+        super(analyzer);
+    }
+
+    @Override
+    protected void parseAndValidate() {
+        processFromTokens();
+    }
+
+    @Override
+    protected TokenContext getTokenContext(Position position) {
+        boolean isInClause = isBetween(getFirstTknIndex(), getLastTknIndex(), position);
+        if( isInClause ) {
+            Token tkn = this.analyzer.getTokenFor(position);
+            return new TokenContext(position, tkn, CONTEXT.FROM_CLAUSE, this);
+        }
+        return null;
+    }
+
+    private void processFromTokens() {
+        // Look for SELECT and FROM tokens
+        Token fromToken = findTokenByKind(FROM);
+        if(fromToken != null ) {
+            setFirstTknIndex(getTokenIndex(fromToken));
+        } else {
+            this.analyzer.addException(new DdlAnalyzerException("Parsing Exception:  there is no 'FROM' keyword in your query expression"));
```

Comment:
It's perfectly valid to use select without from.  People are more familiar with that than using VALUES

Generally it won't define a view, but some customers like to use it for query factoring, such that a common table defines literals:

with constants as (select 1 as val ...)
select ...

Reply:
Due to it's complexity, the query expression completion logic is incomplete and we can work it under [TEIIDTOOLS-992](https://issues.redhat.com/browse/TEIIDTOOLS-992)

The different use case logic for different/simpler forms of the expression can also be added as well to diagnostics as well as completion item framework